src/test/8_Vault.t.sol

```diff
@@ -1624,19 +1622,81 @@ contract vaultTests is Test {
         vault_m.initialize(
             address(this),
             address(mainRegistry),
-            uint8(Constants.UsdNumeraire),
             address(stable),
             address(stakeContract),
             address(interestRateModule)
         );
         assertEq(address(this), vault_m.owner());
 
         vm.startPrank(from);
-        vm.expectRevert("VL: Not factory");
+        vm.expectRevert("VL: You are not the factory");
         vault_m.transferOwnership(to);
         assertEq(address(this), vault_m.owner());
     }
 
+    function testAuthorizeAddress(address toAuth) public {
+        vm.assume(
+            toAuth != address(this) &&
+                toAuth != address(0) &&
+                toAuth != address(factoryContr)
+        );
+        Vault vault_m = new Vault();
+
+        vault_m.authorize(toAuth);
+
+        assertTrue(vault_m.allowed(toAuth));
+
+    }
+
+    function testAuthorizeAddressByNonOwner(address toAuth) public {
+        vm.assume(
+            toAuth != address(this) &&
+                toAuth != address(0) &&
+                toAuth != address(factoryContr)
+        );
+        Vault vault_m = new Vault();
+        address notOwner = address(789);
+
+        vm.startPrank(notOwner);
+        vm.expectRevert("VL: You are not the owner");
+        vault_m.authorize(toAuth);
+        vm.stopPrank();
+
+        assertFalse(vault_m.allowed(toAuth));
+    }      
+
+    function testSetBaseCurrency(address toAuth) public {
+        vm.assume(
+            toAuth != address(this) &&
+                toAuth != address(0) &&
+                toAuth != address(factoryContr)
+        );
+        Vault vault_m = new Vault();
+
+        vault_m.authorize(toAuth);
+
+        
+        vm.startPrank(toAuth);
+        vault_m.setBaseCurrency(Constants.EthBaseCurrency);
+        vm.stopPrank();
+
+        (,,,,,uint256 _baseCurrency) = vault_m.debt();
+        assertEq(_baseCurrency, Constants.EthBaseCurrency);
+    }
+
+    function testSetBaseCurrencyByNonAuthorized() public {
+        Vault vault_m = new Vault();
+        address nonAuthorized = address(789);
+
+        vm.startPrank(nonAuthorized);
+        vm.expectRevert("VL: You are not authorized");
+        vault_m.setBaseCurrency(Constants.EthBaseCurrency);
+        vm.stopPrank();
+
+        (,,,,,uint256 _baseCurrency) = vault_m.debt();
+        assertEq(_baseCurrency, Constants.UsdBaseCurrency);
+    }
+
```

Comment:
Add test when vault already has debt (think best to use a cheatcode here, maybe see it with Jasper)

Reply:
I'll add in an extra test but this test is going to change within next two weeks when we change debt to valueLock ysstem?